src/deploy/functions/deploymentPlanner.ts

```diff
@@ -73,12 +74,13 @@ export function calculateRegionalFunctionChanges(
 
       checkForInvalidChangeOfTrigger(fn, haveFn);
 
-      // Remember old environment variables that might have been set with
-      // gcloud or the cloud console.
-      fn.environmentVariables = {
-        ...haveFn.environmentVariables,
-        ...fn.environmentVariables,
-      };
+      if (!previews.dotenv) {
```

Comment:
I wonder about how we should handle the case where someone has set env OOB in today's world. I think this is probably reasonable behavior (so long as we call it out) for a preview flag, but once we're ready to unflag I think it'd be a breaking change to suddenly start wiping out existing env variables.

Perhaps we should only "forget" these if the preview is enabled **and** at least one `.env` file exists locally?